testcode/lib/conio.c

```diff
@@ -120,6 +132,11 @@ void main(void)
                     cputc(i);
                     inpos = (inpos + 1) & 7;
                 }
+#endif
+/* not all targets have waitvblank() */
```

Comment:
Okay, I added joystick input (just incrementing n). I don't understand your remark regarding kbhit() as it wasn't present AND I don't see a reason to add it. It should now be possible to operate the (from my perspective not that useful) program as intended without a keyboard.